src/components/Modal/index.tsx

```diff
@@ -495,7 +509,7 @@ const Modal: React.FC<ModalProps> = (props) => {
                 {title}
               </h5>
 
-              {(onClose && !dontShowXButton) && (
+              {(onClose && !dontShowXButton && !(isLoading && !isLoadingCancelable)) && (
```

Comment:
Let's put this logic outside jsx and describe what's happening here, now that it's getting so complex